lib/Zonemaster/CLI.pm

```diff
@@ -669,37 +669,99 @@ sub run {
 
     if ( $opt_nstimes ) {
         my $zone = Zonemaster::Engine->zone( $domain );
-        my $max = max map { length( "$_" ) } ( @{ $zone->ns }, q{Server} );
+        my %all_nss = %{ Zonemaster::Engine::Nameserver::object_cache };
+        my @child_nss = @{ $zone->ns };
+        my @parent_nss = @{ $zone->parent->ns };
+        my @nss;
+
+        foreach my $ns_name ( keys %all_nss ) {
+            foreach my $ns ( values %{ $all_nss{$ns_name} } ) {
+                push @nss, $ns if scalar @{ $ns->times } > 0;
+            }
+        }
+
+        my %nss_filter;
+        @nss_filter{ ( @child_nss, @parent_nss ) } = undef;
+        my @other_nss = grep { ! exists $nss_filter{$_} } @nss;
 
         if ( $opt_json ) {
-            my @times = ();
-            foreach my $ns ( @{ $zone->ns } ) {
-                push @times, {
-                    'ns'     => $ns->string,
-                    'max'    => 1000 * $ns->max_time,
-                    'min'    => 1000 * $ns->min_time,
-                    'avg'    => 1000 * $ns->average_time,
-                    'stddev' => 1000 * $ns->stddev_time,
-                    'median' => 1000 * $ns->median_time,
-                    'total'  => 1000 * $ns->sum_time
+            my @times;
+
+            sub json_nstimes {
+                my ( $ns ) = @_;
+                return {
+                    'ns'      => $ns->string,
+                    'max'     => 1000 * $ns->max_time,
+                    'min'     => 1000 * $ns->min_time,
+                    'avg'     => 1000 * $ns->average_time,
+                    'stddev'  => 1000 * $ns->stddev_time,
+                    'median'  => 1000 * $ns->median_time,
+                    'total'   => 1000 * $ns->sum_time,
+                    'count'   => scalar @{ $ns->times }
                 };
             }
```

Comment:
Nested `sub`s aren’t officially supported in Perl. Either use [`my sub`](https://perldoc.perl.org/5.26.0/perlsub#Lexical-Subroutines) with `use v5.26` or move it to the top-level of the file.